js/ui/windowManager.js

```diff
@@ -1455,9 +1455,18 @@ const WindowManager = new Lang.Class({
     },
 
     _endlessCodingLaunched : function(actor, event, window) {
-        let tracker = Shell.WindowTracker.get_default();
-        tracker.track_coding_app_window(window);
-        Util.trySpawn(['flatpak', 'run', 'org.gnome.Builder', '-s']);
+        let currentSession = this._currentEndlessCodingSessions.filter(function(session) {
+            return session.windowApp === window;
+        });
+        if (currentSession.length === 0)
+            return;
+        if (currentSession[0].windowBuilder === undefined) {
+            let tracker = Shell.WindowTracker.get_default();
+            tracker.track_coding_app_window(window);
```

Comment:
The rest of the code is the same, I would keep it that way, I think is better readable as well.